pocs/camera/camera.py

```diff
@@ -707,6 +705,30 @@ def _setup_observation(self, observation, headers, filename, **kwargs):
         # be override by passed parameter so update here.
         metadata['exptime'] = exptime
 
+        # Prepare the filterwheel
+        if self.filterwheel is not None:
+
+            # Check that the filterwheel is not moving
+            if self.filterwheel.is_moving:
+
+                msg = f'Attempt to prepare observation on {self} while' \
+                        ' filterwheel is moving, ignoring.'
+                self.logger.error(msg)
+                raise error.PanError(msg)
+
+            # Move filterwheel if necessary
+            if observation.filter_name is not None:
+
+                try:
+                    self.filterwheel.move_to(observation.filter_name, blocking=True)
+                except Exception as e:
+                    self.logger.error(f'Error moving filterwheel on {self} to' \
+                                      f' {observation.filter_name}: {e}')
+                    raise(e)
+
+                # Store the filter name in metadata
+                metadata['filter_name'] = observation.filter_name
```

Comment:
I _think_ this is also redundant.... The current filter is always written into the FITS headers of any exposure taken with a POCS camera. I don't think that manually inserting it into the `metadata` `dict` here adds anything.

Reply:
Does it? Seems like the headers get added in `AbstractCamera._process_fits` using `info=metadata` passed to `AbstractCamera._process_exposure`. I can't see any other way filter_name (which is a new attribute of Observation) is written to the headers.

Reply:
`._process_fits()` adds additional entries to FITS headers that relate to the details of the observation. The headers have already been created and populated with all the information about camera hardware state by the `._create_fits_headers()` methods of the camera classes and the `._add_fits_keywords()` methods of the focuser and filterwheel classes (remember that `.take_exposure()` is supposed to useful by itself, that's why it doesn't rely on `._process_fits` for this).

One of the keywords that `._create_fits_headers()` adds is `FILTER`, which is set from the camera class `.filter_type` property, and if a camera has a filterwheel that property will get its value from the filterwheel's `current_filter` property.  If the camera doesn't have a filterwheel this value comes either from the camera config for fixed bandpass filter (that's how it's worked for Huntsman up to now) or, failing that, from the camera itself (things like 'M' for monochrome camera or 'RGGB' for colour cameras).

So, the FITS headers will already contain the filter that was used to take the exposure. The question is whether it is useful to separately include the requested filter name from the observation. In general these will be different, either because the camera doesn't have a filterwheel so can't use a matching filter, or because the filter name that the filterwheel is using contains additional information (e.g. batch/serial number) after the band name.